.github/workflows/ci.yml

```diff
@@ -995,6 +995,23 @@ jobs:
         run: cargo check-external-types --all-features
         working-directory: tokio
 
+  check-unexpected-lints-cfgs:
+    name: check unexpected lints and cfgs
+    needs: basics
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust ${{ env.rust_nightly }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust_nightly }}
+      - name: don't allow warnings
+        run: sed -i '/#!\[allow(unknown_lints, unexpected_cfgs)\]/d' */src/lib.rs
```

Comment:
I don't think you need this part, the `-F` (forbid) flag cannot ever be overridden. Including inside the source code.

<details>

```text
error[E0453]: allow(unexpected_cfgs) incompatible with previous forbid
 --> src/lib.rs:1:25
  |
1 | #![allow(unknown_lints, unexpected_cfgs)]
  |                         ^^^^^^^^^^^^^^^ overruled by previous forbid
  |
  = note: `forbid` lint level was set on command line
```

</details>

Reply:
I added it because adding the `-F` flag causes compilation to fail when it encounters the `#[allow]`.